predictive_test_selection/collect_test_failure_data.py

```diff
@@ -0,0 +1,168 @@
+import logging
+import os
+import subprocess
+import time
+import xml.etree.ElementTree as ET
+from datetime import datetime
+from enum import IntFlag, auto
+from typing import Any, List, Mapping, Sequence, Tuple
+
+import requests
+
+LOGGER = logging.getLogger(__name__)
+
+
+class CommonFileExtension(IntFlag):
```

Comment:
I am curious what the impact is if we at some point decide to add extra extensions. Does historical data we collected become useless then?

Reply:
I think that's a good idea. That way decouple the data collection part from the model generation part, which gives us more flexibility. And I guess that the store overhead on the database will be limited.